### PR TITLE
Fix response processing being blocked

### PR DESCRIPTION
**Pull request by FooIbar**

### CODE REVIEW

File: app/src/main/kotlin/com/hippo/ehviewer/ktor/CronetEngine.kt [hunk 1]
```diff
@@ -65,11 +62,11 @@ object CronetEngine : HttpClientEngineBase("Cronet") {
                     info.toHttpResponseData(
                         requestTime = requestTime,
                         callContext = callContext,
-                        responseBody = GlobalScope.writer(callContext) {
-                            pool.useInstance {
-                                chunkChan = Channel()
-                                request.read(it)
-                                chunkChan.consumeEach { buffer ->
+                        // Don't use the same coroutine context as the response, as it will block response processing.
+                        responseBody = writer {
```
Comment (revonateB0T) [inline app/src/main/kotlin/com/hippo/ehviewer/ktor/CronetEngine.kt:66]: This `WriterJob` will never be cancelled if request is cancelled when processing processing

File: app/src/main/kotlin/com/hippo/ehviewer/ktor/CronetEngine.kt [hunk 2]
```diff
@@ -65,11 +62,11 @@ object CronetEngine : HttpClientEngineBase("Cronet") {
                     info.toHttpResponseData(
                         requestTime = requestTime,
                         callContext = callContext,
-                        responseBody = GlobalScope.writer(callContext) {
```
Comment (revonateB0T) [inline app/src/main/kotlin/com/hippo/ehviewer/ktor/CronetEngine.kt:68]: I suggest we just `+ Dispatchers.IO`

**Review by revonateB0T — APPROVED** [review]

File: app/src/main/kotlin/com/hippo/ehviewer/ktor/CronetEngine.kt (hunk 1 above)
Comment (revonateB0T) [reply app/src/main/kotlin/com/hippo/ehviewer/ktor/CronetEngine.kt:66]: > ~This `WriterJob` will never be cancelled if request is cancelled when processing response~ It doesn't matter as both `CronetRequest` and `KtorRequest` will be released so no res-leak

`DirectByteBufferPool` instance will leak

**Comment by revonateB0T** [comment]
https://github.com/FooIbar/EhViewer/pull/972

**Comment by revonateB0T** [comment]
FWIW this problem should exist in other `HttpEngine` as well
Maybe we should address it with a custom `execute` function where we `finally` cancel the `callContext`

**Comment by revonateB0T** [comment]
Anyway I think this is a internal misbehaviour in ktor
https://github.com/ktorio/ktor/blob/46486b50d5248a7c0c05e52430948cbc1e82f000/ktor-client/ktor-client-core/common/src/io/ktor/client/statement/HttpStatement.kt#L134
This should be `cancelAndJoin` as all about that request is meaningless after `execute` that `block`

**Comment by FooIbar** [comment]
#988

**Pull request closed without merging**